MoreLinq/StartsWith.cs

```diff
@@ -0,0 +1,49 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2012 Atif Aziz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        ///     Determines whether the beginning of <paramref name="first" /> matches <paramref name="second" />.
+        /// </summary>
+        /// <typeparam name="T1">Type of elements in <paramref name="first" />.</typeparam>
+        /// <typeparam name="T2">Type of elements in <paramref name="second" />.</typeparam>
+        /// <param name="first">The sequence to check.</param>
+        /// <param name="second">The sequence to compare to.</param>
+        /// <returns>
+        ///     <c>true</c> if <paramref name="first" /> begins with elements equivalent to <paramref name="second" />.
+        /// </returns>
+        /// <remarks>
+        ///     This is the <see cref="IEnumerable{T}" /> equivalent of <see cref="string.StartsWith(string)" /> and
+        ///     it calls <see cref="object.Equals(object)" /> on pairs of elements at the same index.
+        /// </remarks>
+        public static bool StartsWith<T1, T2>(this IEnumerable<T1> first, IEnumerable<T2> second)
```

Comment:
I was contemplating the very same thing, and originally started out with only one.
What I figured was that we limit the number of usecases by doing only 1 type, since object.Equals() can be called on any two types, and in some usecases that may make sense.

So I was going to write an example how it could be convenient to allow comparing a sequence of ints to a sequence of longs, but then I tested 1L.Equals(1) and realized it is in fact false. I was not aware of this. Reading up on Equals(), it's generally assumed that if they are not of the same type, it should return false. The `==` operator, however, does return true, but that would not apply in this implementation anyway.

Long story short, yes, you are absolutely right. It should be only one type. I'll update the PR.